src/components/StatusSelector.tsx

```diff
@@ -0,0 +1,146 @@
+import { useState } from 'react'
+import { Button, Form, Modal } from 'react-bootstrap'
+import { Link } from 'react-router-dom'
+import { ButtonReview } from './CustomButton'
+import { StatusType, ComponentType, StatusSelectorInterface } from '../interfaces/booking.interfaces'
+import { RoutePath } from '../interfaces/route.interface'
+import { useTokenContext } from '../hooks/CustomProvider'
+
+/**
+ * Return various React.FC<StatusSelectorInterface> which is separated by component
+ * @param status StatusType of booking
+ * @param component ComponentType of component
+ * @param locationId identifier of location
+ * @returns ComponentType.PROGRESS_CIRCLE which is separated by ComponentType
+ *          ComponentType.ACTION_BUTTON which is separated by ComponentType and locationId
+ */
+export const StatusSelector: React.FC<StatusSelectorInterface> = ({
+  status,
+  component,
+  locationId,
+  providerId,
+  price,
+}) => {
+  const { currentToken } = useTokenContext()
+  const [error, setError] = useState<string | null>(null)
+  const [show, setShow] = useState(false)
+  const [takeReceipt, setTakeReceipt] = useState<boolean>(false)
+
+  // fetch payment page with stripe API
+  const fetchPaymentPage = async (
+    provider_id: string,
+    location_id: string,
+    price: number,
+    takeReceipt: boolean,
+  ) => {
+    try {
+      const url = `http://${URL}/stripe/create-checkout-session`
+
+      const bodyData = {
+        provider_id: provider_id,
+        location_id: location_id,
+        price: price,
+        takeReceipt: takeReceipt,
+      }
```

Comment:
I can't make a booking arhhhh.

not sure that a problem come from me or request body
![image](https://user-images.githubusercontent.com/33095457/229573627-30ea51ef-93ca-4fd9-a7ca-6d47d34a2c05.png)

![image](https://user-images.githubusercontent.com/33095457/229573574-4bb0a83b-cbb4-42e4-93f1-a8476c717c98.png)